src/Microsoft.SqlTools.ServiceLayer/DacFx/GetObjectsFromTSqlModelOperation.cs

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.SqlServer.Dac.Model;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent request to get objects from model
+    /// </summary>
+    class GetObjectsFromTSqlModelOperation
+    {
+        public TSqlModel Model;
+        public GetObjectsFromTSqlModelParams Parameters { get; }
+
+        public GetObjectsFromTSqlModelOperation(GetObjectsFromTSqlModelParams parameters, TSqlModel model)
+        {
+            Validate.IsNotNull("parameters", parameters);
```

Comment:
Any reason you're not checking the model as well? (even if you check it before it's used in the code, we shouldn't be assuming that's the only place it'll ever be used. Best to be consistent with all our validations.)